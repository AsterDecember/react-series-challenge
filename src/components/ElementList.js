import React from 'react';
import Element from './Element';

const ElementList = (props) => {
    console.log('test test :',props);
    const searchGifsLength = props.search ? Object.keys(props.search).length : 0;

    const renderGifs = () => {
        const gifs = searchGifsLength > 0 ? props.search : props.all;
        return  gifs ? Object.keys(gifs).map((gif) => <Element key={gif.id} gif={gifs[gif]}> x</Element>) : <h1>No info</h1>
    };

    return (
        <div>
            {renderGifs()}
        </div>
    );
}


export default ElementList;