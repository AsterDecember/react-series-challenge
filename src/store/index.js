import { createStore, applyMiddleware, compose } from 'redux';
import combineReducers from '../reducers';
import createSagaMiddleware from 'redux-saga';
import rootSaga from '../sagas';

const initialState   = {};
const sagaMiddleware = createSagaMiddleware();

const middleware = [
    sagaMiddleware
];

const composedEnhancers = compose(
    applyMiddleware(...middleware)
);

const index = createStore(
    combineReducers,
    initialState,
    composedEnhancers
);
sagaMiddleware.run(rootSaga);
export default index;