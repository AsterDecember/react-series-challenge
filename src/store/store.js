import { createStore } from 'redux';
import combineReducers from './reducers';

const initialState = {};

const store = createStore(
    combineReducers,
    initialState
);

export default store;