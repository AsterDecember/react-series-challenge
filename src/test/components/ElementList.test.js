import React from 'react';
import { Provider } from 'react-redux';
import ElementList from '../../components/ElementList';
import renderer from 'react-test-renderer';
import store from '../../store/index';
import localStorageMock from './localStorageMock';

describe('ElementList', () => {

    it('should render without problems', () => {
        const wrapper = renderer.create(
            <Provider store={store}>
                <ElementList />
            </Provider>
        ).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

});
