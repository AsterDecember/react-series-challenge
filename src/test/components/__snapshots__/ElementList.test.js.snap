// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementList should render without problems 1`] = `
<div
  className="sc-EHOje gqTaLh"
>
  <h1>
    No info
  </h1>
</div>
`;
